### PR TITLE
357404

### LINKED ISSUE #3295: 357404
GDoc: 

Category: Immunofluorescence

### PR DESCRIPTION
**Pull request by eskila**
Histology slide immunostaining with custom Shandon coverplate/slide blocks (holders with "wells" for pipetting)
closes #3295

### CODE REVIEW

File: protocols/357404/357404.ot2.apiv2.py [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+
+        def flash_lights(self):
+            """
+            Flash the lights of the robot to grab the users attention
+            """
+            initial_light_state = ctx.rail_lights_on
+            opposite_state = not initial_light_state
+            for _ in range(5):
+                ctx.set_rail_lights(opposite_state)
+                time.sleep(0.5)
```
Comment (chazmanc) [inline protocols/357404/357404.ot2.apiv2.py:277]: ctx.delay(0.5) instead of time.sleep(0.5)

Comment (eskila) [reply protocols/357404/357404.ot2.apiv2.py:277]: Why didn't I think of that! thanks

File: protocols/357404/357404.ot2.apiv2.py [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+    # Set the temperature to 4 degrees on the temperature module.
+    temp_mod.set_temperature(4)
+    # Transfer 100 µL of block from the tuberack to each destination well
+    # (slide) (could be done as a multi-dispense with the P1000)
+    # Measure time from the start of pipetting and subtract from 1 hr pause
+    # so that the next pipetting step starts after 1 hour and not 1 hour +
+    # the time it takes to finish the reagent transfer
+    if not is_start_after_1st_incbn:  # Skip 1st incubation?
+        ctx.comment("\n\nAdding block reagent\n")
+        t = time.time()
```
Comment (chazmanc) [inline protocols/357404/357404.ot2.apiv2.py:455]: @StevenPlonk developed a good solution to this problem (set incubation times dependent on pipetting steps), but something still in progress...

File: protocols/357404/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+### Labware
+* [Opentrons tuberacks](https://shop.opentrons.com/4-in-1-tube-rack-set/)
+* [Opentrons tubes & vials](https://shop.opentrons.com/consumables/)
+* [Agilent 1-Well Reservoir 290 mL](https://labware.opentrons.com/agilent_1_reservoir_290ml)
+* [1000 uL tipracks](https://shop.opentrons.com/opentrons-1000-l-tips/)
+* [300 uL tipracks](https://shop.opentrons.com/opentrons-300ul-tips-1000-refills/)
+
+### Pipettes
+* [P300 single-Channel (GEN2)}](https://shop.opentrons.com/single-channel-electronic-pipette-p20/)
+* [P1000 single-Channel (GEN2)}](https://shop.opentrons.com/single-channel-electronic-pipette-p20/)
```
Comment (chazmanc) [inline protocols/357404/README.md:43]: remove the `}` at the end of the text

**Review by chazmanc — COMMENTED** [review]
all instances of time() need to be replaced.

**Comment by eskila** [comment]
I replaced all the time() calls with `time per block` parameter the user will have to set manually.

**Pull request merged**